docs/sql/backup_restore.txt

```diff
@@ -0,0 +1,174 @@
+.. highlight:: psql
+.. _backup-restore:
+
+Backup and Restore
+==================
+
+Backup
+------
+
+In CRATE Backups are called *Snapshots*. They represent the state of the tables in a CRATE cluster
+at the time the *Snapshot* was created.
+A *Snapshots* is always stored in a *Repository* which has to be created up front:
+
+Creating a Repository
+~~~~~~~~~~~~~~~~~~~~~
+
+Repositories are used to store, manage and restore *snapshots*.
+
+They are created using the :ref:`ref-create-repository` statement::
+
+    cr> CREATE REPOSITORY where_my_snapshots_go TYPE fs
+    ... WITH (location='repo_path', compress=true);
+    CREATE OK (... sec)
+
+Repositories are uniquely identified by their name. Every repository has a specific
+type which determines how snapshots are stored.
+
+CRATE supports different repository types:  :ref:`ref-create-repository-types-fs`, :ref:`ref-create-repository-types-hdfs`,
+and :ref:`ref-create-repository-types-url`. Support for further types might be added using :ref:`plugins`.
+
+Repository creation simply configures it inside the crate cluster. In general no
+data is written, no snapshots inside repositories are changed or deleted.
+This way you can tell the CRATE cluster about repositories that already contain snapshots.
+
+Creating a repository with the same name will result in an error::
+
+    cr> CREATE REPOSITORY where_my_snapshots_go TYPE fs
+    ... WITH (location='another_repo_path', compress=false);
+    ERROR
+
+Creating a Snapshot
+~~~~~~~~~~~~~~~~~~~
+
+Snapshots are backups of the current state of the tables in the CRATE cluster.
+They are created inside a repository and can contain any number of tables.
+The :ref:`ref-create-snapshot` statement is used to create a snapshots::
+
+    cr> CREATE SNAPSHOT where_my_snapshots_go.snapshot1 ALL
+    ... WITH (wait_for_completion=true, partial=true, ignore_unavailable=true);
+    CREATE OK (... sec)
+
+A snapshot is referenced by the name of the repository and the snapshot name, separated by a dot.
+If used with ``ALL``, all tables of the cluster are stored inside the snapshot
+(except :ref:`sys-tables <system-information>`, :ref:`information_schema` tables and blob tables).
```

Comment:
maybe we should mention that all persistent settings and the full cluster metadata is saved as well using ALL